test-suite/src/test/scala/org/scalajs/testsuite/javalib/math/BigDecimalConstructorsTest.scala

```diff
@@ -0,0 +1,479 @@
+      expect(aNumber.unscaledValue() == bA).toBeTruthy
+      expect(aScale).toEqual(aNumber.scale())
+    }
+
+    it("testConstrStringWithExponentWithoutPoint5") {
+      val a = "238768787678287E214"
+      val aScale = -214
+      val bA = new BigInteger("238768787678287")
+      val aNumber = new BigDecimal(a)
+      expect(aNumber.unscaledValue() == bA).toBeTruthy
+      expect(aScale).toEqual(aNumber.scale())
+    }
+
+    it("testConstrStringWithExponentWithPoint1") {
+      val a = "23985439837984782435652424523876878.7678287e+214"
+      val aScale = -207
+      val bA = new BigInteger("239854398379847824356524245238768787678287")
+      val aNumber = new BigDecimal(a)
+      expect(aNumber.unscaledValue() == bA).toBeTruthy
+      expect(aScale).toEqual(aNumber.scale())
+    }
+
+    it("testConstrStringWithExponentWithPoint2") {
+      val a = "238096483923847545735673567457356356789029578490276878.7678287e-214"
+      val aScale = 221
+      val bA = new BigInteger("2380964839238475457356735674573563567890295784902768787678287")
+      val aNumber = new BigDecimal(a)
+      expect(aNumber.unscaledValue() == bA).toBeTruthy
+      expect(aScale).toEqual(aNumber.scale())
+    }
+
+    it("testConstrStringWithExponentWithPoint3") {
+      val a = "2380964839238475457356735674573563567890.295784902768787678287E+21"
+      val aScale = 0
+      val bA = new BigInteger("2380964839238475457356735674573563567890295784902768787678287")
+      val aNumber = new BigDecimal(a)
+      expect(aNumber.unscaledValue() == bA).toBeTruthy
+      expect(aScale).toEqual(aNumber.scale())
+    }
+
+    it("testConstrStringWithExponentWithPoint4") {
+      val a = "23809648392384754573567356745735635678.90295784902768787678287E+21"
+      val aScale = 2
+      val bA = new BigInteger("2380964839238475457356735674573563567890295784902768787678287")
+      val aNumber = new BigDecimal(a)
+      expect(aNumber.unscaledValue() == bA).toBeTruthy
+      expect(aScale).toEqual(aNumber.scale())
+    }
+
+    it("testConstrStringWithExponentWithPoint5") {
+      val a = "238096483923847545735673567457356356789029.5784902768787678287E+21"
+      val aScale = -2
+      val bA = new BigInteger("2380964839238475457356735674573563567890295784902768787678287")
+      val aNumber = new BigDecimal(a)
+      expect(aNumber.unscaledValue() == bA).toBeTruthy
+      expect(aScale).toEqual(aNumber.scale())
+    }
+
+    it("testConstrStringWithLeadingZeros") {
+      expect(1).toEqual(new BigDecimal("-000.1").precision())
+      expect(4).toEqual(new BigDecimal("001234").precision())
+      expect(4).toEqual(new BigDecimal("-5555").precision())
+      expect(1).toEqual(new BigDecimal("0").precision())
+      expect(1).toEqual(new BigDecimal("-0").precision())
+      expect(4).toEqual(new BigDecimal("001234e3").precision())
+      expect(5).toEqual(new BigDecimal("00056789e+17").precision())
+      expect(6).toEqual(new BigDecimal("0900000e-42").precision())
```

Comment:
Left and right should be reversed. Actual is in the call to `expect`, expected is in the call to `toEqual`.
There might be other occurrences of this problem.